data-prepper-api/src/test/java/org/opensearch/dataprepper/model/failures/DlqObjectTest.java

```diff
@@ -0,0 +1,267 @@
+        pluginName = randomUUID().toString();
+        pipelineName = randomUUID().toString();
+        failedData = randomUUID();
+    }
+
+    @Test
+    public void test_build_with_timestamp() {
+
+        final DlqObject testObject = DlqObject.builder()
+                .withPluginId(pluginId)
+                .withPluginName(pluginName)
+                .withPipelineName(pipelineName)
+                .withFailedData(failedData)
+                .withTimestamp(randomUUID().toString())
+                .build();
+
+        assertThat(testObject, is(notNullValue()));
+    }
+
+    @Test
+    public void test_build_without_timestamp() {
+
+        final DlqObject testObject = DlqObject.builder()
+            .withPluginId(pluginId)
+            .withPluginName(pluginName)
+            .withPipelineName(pipelineName)
+            .withFailedData(failedData)
+            .build();
+
+        assertThat(testObject, is(notNullValue()));
+    }
+
+    @Nested
+    public class InvalidBuildParameters {
+
+        private void createTestObject() {
+            DlqObject.builder()
+                .withPluginId(pluginId)
+                .withPluginName(pluginName)
+                .withPipelineName(pipelineName)
+                .withFailedData(failedData)
+                .build();
+        }
+        @Test
+        public void test_invalid_pluginId() {
+            pluginId = null;
+            assertThrows(NullPointerException.class, this::createTestObject);
+            pluginId = "";
+            assertThrows(IllegalArgumentException.class, this::createTestObject);
+        }
+
+        @Test
+        public void test_invalid_pluginName() {
+            pluginName = null;
+            assertThrows(NullPointerException.class, this::createTestObject);
+            pluginName = "";
+            assertThrows(IllegalArgumentException.class, this::createTestObject);
+        }
+
+        @Test
+        public void test_invalid_pipelineName() {
+            pipelineName = null;
+            assertThrows(NullPointerException.class, this::createTestObject);
+            pipelineName = "";
+            assertThrows(IllegalArgumentException.class, this::createTestObject);
+        }
+
+        @Test
+        public void test_invalid_failedData() {
+            failedData = null;
+            assertThrows(NullPointerException.class, this::createTestObject);
+        }
+    }
+
+    @Nested
+    class Getters {
+
+        private DlqObject testObject;
+
+        @BeforeEach
+        public void setup() {
+
+            testObject = DlqObject.builder()
+                .withPluginId(pluginId)
+                .withPluginName(pluginName)
+                .withPipelineName(pipelineName)
+                .withFailedData(failedData)
+                .build();
+        }
+
+        @Test
+        public void test_get_pluginId() {
+            final String actualPluginId = testObject.getPluginId();
+            assertThat(actualPluginId, is(notNullValue()));
```

Comment:
These lines are unnecessary. If the value is `null`, the `assertThat` below will clearly state it is null.